src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_Warnings.cs

```diff
@@ -792,7 +792,7 @@ private void CheckLiftedBinOp(BoundBinaryOperator node)
 
                     string always = node.OperatorKind.Operator() == BinaryOperatorKind.NotEqual ? "true" : "false";
 
-                    if (_compilation.FeatureStrictEnabled || !node.OperatorKind.IsUserDefined())
+                    if (_compilation.FeatureStrictEnabled || !node.OperatorKind.IsUserDefined() || ErrorCode.WRN_NubExprIsConstBool2.GetWarningVersion() <= this._compilation.Options.WarningVersion)
```

Comment:
>FeatureStrictEnabled [](start = 37, length = 20)

Should we assume that `FeatureStrictEnabled` is true when warn version >=5?

Reply:
No.  `FeatureStrictEnabled` is orthogonal and most of the diagnostics under its control are currently errors.  Once we migrate something to a warning wave we can remove it from `FeatureStrictEnabled` after a period of time sufficient for people to start using it.

---
In reply to: [452495746](https://github.com/dotnet/roslyn/pull/45750#discussion_r452495746) [](ancestors = 452495746)